src/checkpoint/bottomup.rs

```diff
@@ -35,11 +37,9 @@ pub struct NativeChildCheck {
 
 /// The trait that handles the bottom up checkpoint submission data preparation and actual submission.
 #[async_trait]
-pub trait BottomUpHandler: Send + Sync + VoteQuery<NativeBottomUpCheckpoint> {
-    /// Get the checkpoint period
-    async fn checkpoint_period(&self, subnet_id: &SubnetID) -> Result<ChainEpoch>;
-    /// Get the list of validators in the subnet id
-    async fn validators(&self, subnet_id: &SubnetID) -> Result<Vec<Address>>;
+pub trait BottomUpHandler:
+    Send + Sync + VoteQuery<NativeBottomUpCheckpoint> + CheckpointUtilQuery<NativeBottomUpCheckpoint>
```

Comment:
Can we merge `VoteQuery` and `CheckpointUtilQuery` into the same trait `CheckpointQuery`? Even if this is not possible, I think we should rename `CheckpointUtilQuery` to `CheckpointQuery`.